src/Datadog.Trace/Tracer.cs

```diff
@@ -119,6 +119,12 @@ internal Tracer(TracerSettings settings, ITraceWriter traceWriter, ISampler samp
             _scopeManager = scopeManager ?? new AsyncLocalScopeManager();
             Sampler = sampler ?? new RuleBasedSampler(new RateLimiter(Settings.MaxTracesSubmittedPerSecond));
 
+            // W3C propagator requires Otel TraceId convention as it's specification clearly states lengths of traceId and spanId values in the header.
+            if (Settings.Propagator == PropagatorType.W3C && Settings.Convention != ConventionType.OpenTelemetry)
```

Comment:
Shouldn't it be inside ContextPropagatorBuilder?  So the builder takes care of validation and building and logic remains visible for unit tests as well.

like that >>
```
if (propagator == PropagatorType.W3C && traceIdConvention is not OtelTraceIdConvention)
...
```